app/Http/Controllers/QuizAPIController.php

```diff
@@ -51,4 +49,19 @@ function addQuiz(Request $request)
             ], 500);
         }
     }
+
+    function singleQuiz(int $id)
+    {
+        try {
+            $quizzes = Quiz::with(['questions' => ['answers'],])->find($id);
+            return response()->json([
+                'message' => 'Quiz retrieved',
+                'data' => $quizzes
+            ], 200);
+        } catch (Exception $e) {
+            return response()->json([
+                'message' => 'Quiz not found'
+            ], 404);
+        }
```

Comment:
The fault caught by `try {...} catch {...}` is not that it didn't find a quiz. 

You are `try`ing a database operation that may fail. If your database fails that is a different thing to not finding a quiz.